src/Form/CoyoteImgBatchForm.php

```diff
@@ -115,39 +117,67 @@ private function hasValidBatchProcessingConfig(): bool {
   } 
 
   public function submitForm(array &$form, FormStateInterface $form_state): void {
-     $nids = \Drupal::entityQuery('node')->execute();
+     $entities = array_keys(\Drupal::entityTypeManager()->getDefinitions());
      $operations = [];
 
-     foreach ($nids as $nid){     
-       $operations[] = [['\Drupal\coyote_img_desc\Form\CoyoteImgBatchForm','batchProcessingNodes'], [$nid]];
+     foreach ($entities as $entity){  
+        $ids = \Drupal::entityQuery($entity)->execute();
+        foreach ($ids as $id){     
+          $operations[] = [['\Drupal\coyote_img_desc\Form\CoyoteImgBatchForm','batchProcessingEntities'], [$id, $entity]];
+        }
      }
      $batch = [
-          'title' => $this->t('Batch processing All Nodes ...'),
+          'title' => $this->t('Batch processing All Images ...'),
           'operations' => $operations,
-          'finished' => ['Drupal\coyote_img_desc\Form\CoyoteImgBatchForm','batchProcessingNodesFinished'],
-      ];
-      batch_set($batch);
+          'finished' => ['Drupal\coyote_img_desc\Form\CoyoteImgBatchForm','batchProcessingEntitiesFinished'],
+     ];
+
+     batch_set($batch);
   }
   
-  public function batchProcessingNodes($nid, &$context) {
-     $message = 'Proccessing ALL nodes to coyote';
-
+  public static function batchProcessingEntities($id, $entity, &$context) {
+     $message = 'Proccessing ALL entities to coyote';
      $results = $context['results'];
    
      $config = \Drupal::config('coyote_img_desc.settings');
 
-     $entity_type = 'node';
+     $entity_type = $entity;
      $view_mode = 'default';
+     $display = true;
+
+     try {
+
+        $viewBuilder = \Drupal::entityTypeManager()->getViewBuilder($entity_type);
+        $storage = \Drupal::entityTypeManager()->getStorage($entity_type);
+        $s = $storage->load($id);
 
-     $viewBuilder = \Drupal::entityTypeManager()->getViewBuilder($entity_type);
-     $storage = \Drupal::entityTypeManager()->getStorage($entity_type);
-     $node = $storage->load($nid);
-     $build = $viewBuilder->view($node, $view_mode);
-     $build['_coyote_node_url'] = $node->toUrl('canonical', ['absolute' => true])->toString();
-     $hostUri = $node->toUrl('canonical', ['absolute' => true])->toString();
-     $output = render($build);
+        $build = $viewBuilder->view($s, $view_mode);
+
+        $hostUri = $s->toUrl('canonical', ['absolute' => true])->toString();
+    
+        $vm = "";
+        if (isset($build["#view_mode"])) { $vm .= $build["#view_mode"]; }
+           \Drupal::logger('coyote')->notice("4".print_r($hostUri,true). " ".print_r(array_keys($build),true). " vm:".$vm);
+          $output = render($build);
+     }
+     catch (UndefinedLinkTemplateException $e) {
```

Comment:
This can be simplified using `catch(ExceptionA | ExceptionB | ExceptionC)` although further dow you're already catching `Exception` as a general catch as well. Is this intended?

Reply:
Is `$vm` used anywhere else?

Reply:
> This can be simplified using `catch(ExceptionA | ExceptionB | ExceptionC)` although further dow you're already catching `Exception` as a general catch as well. Is this intended?

Yes, is intended. Strangely enough these Exceptions are not caught with Exception $e. 
I started with treating all 3 differently, but that was not needed, so will use the | coding.